address-space-controller/src/main/java/io/enmasse/controller/EndpointController.java

```diff
@@ -30,22 +33,142 @@
+
+        List<EndpointSpec> endpoints = validateEndpoints(addressSpaceResolver, addressSpace);
+
+        endpoints = removeAddressSpaceConsoleEndpoint(infraUuid, endpoints, addressSpace);
+
+        // Ensure the required certs are set
+        List<EndpointSpec> newEndpoints = new ArrayList<>();
+        for (EndpointSpec endpoint : endpoints) {
+            EndpointSpecBuilder endpointBuilder = new EndpointSpecBuilder(endpoint);
+
+            CertSpecBuilder certSpec = endpoint.getCert() != null ? new CertSpecBuilder(endpoint.getCert()) : new CertSpecBuilder();
+            if (certSpec.getProvider() == null) {
+                certSpec.withProvider(defaultCertProvider);
+            }
+
+            if (certSpec.getSecretName() == null) {
+                certSpec.withSecretName(KubeUtil.getExternalCertSecretName(endpoint.getService(), addressSpace));
+            }
+
+            endpointBuilder.withCert(certSpec.build());
+            newEndpoints.add(endpointBuilder.build());
+        }
+
+        addressSpace = new AddressSpaceBuilder(addressSpace)
+                .editOrNewSpec()
+                .withEndpoints(newEndpoints)
+                .endSpec()
+                .build();
+        return addressSpace;
+    }
+
+    private List<EndpointSpec> removeAddressSpaceConsoleEndpoint(String infraUuid, List<EndpointSpec> endpoints, AddressSpace addressSpace) {
+        List<EndpointSpec> copy = new ArrayList<>(endpoints);
+
+        // Remove any address-space-specific console endpoint.
+        for (Iterator<EndpointSpec> iterator = copy.iterator(); iterator.hasNext(); ) {
+            EndpointSpec endpoint = iterator.next();
+
+            if ("console".equals(endpoint.getName())) {
+                iterator.remove();
+
+                if (endpoint.getService() != null) {
+                    String serviceName = KubeUtil.getAddressSpaceServiceName(endpoint.getService(), addressSpace);
+                    Service service = client.services().inNamespace(namespace).withName(serviceName).get();
+                    if (service != null && service.getMetadata().getLabels() != null && infraUuid.equals(service.getMetadata().getLabels().get(LabelKeys.INFRA_UUID))) {
+                        client.services().inNamespace(namespace).delete(service);
+                    }
+                }
+
+                CertSpec cert = endpoint.getCert();
+                if (cert != null && "selfsigned".equals(cert.getProvider()) && cert.getSecretName() != null) {
+                    Secret secret = client.secrets().inNamespace(namespace).withName(cert.getSecretName()).get();
+                    if (secret != null && secret.getMetadata().getLabels() != null && infraUuid.equals(secret.getMetadata().getLabels().get(LabelKeys.INFRA_UUID))) {
+                        client.secrets().inNamespace(namespace).delete(secret);
+                    }
+                }
+
+                if (isOpenShift) {
```

Comment:
I believe that in the case of !isOpenShift, there is an console-$uuid-external service that should be removed as well.

Reply:
Ah.. I am not handling loadbalancer case.. is this what you refer to?  I'll fix.

Reply:
Yes

Reply:
Hopefully resolved by my last push.  I had to give responsibilities back to the CreateController as the remainder of the controller needed certificate references.

I've manually tested the upgrade case for both standard/brokered and confirmed that the console resources are removed.